### PR TITLE
DBZ-1040 Metrics for lag behind source, offset and tx count

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1040

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/pipeline/metrics/StreamingChangeEventSourceMetrics.java [hunk 1]
```diff
@@ -36,4 +50,52 @@ public boolean isConnected() {
...
+    @Override
+    public void onEvent(DataCollectionId source, OffsetContext offset, Object key, Struct value, EventMetadataProvider metadataProvider) {
+        super.onEvent(source, offset, key, value, metadataProvider);
+
+        final long eventTimestamp = metadataProvider.getEventTimestamp(source, offset, key, value);
+        if (eventTimestamp != -1) {
+            secondsBehindSource.set((System.currentTimeMillis() - eventTimestamp) / 1000);
+        }
+
+        final String transactionId = metadataProvider.getTransactionId(source, offset, key, value);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/metrics/StreamingChangeEventSourceMetrics.java]: The TX id itself should be exposed, too.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReaderMetrics.java [hunk 2]
```diff
@@ -138,4 +141,18 @@ public void onLargeTransaction() {
         return schema.monitoredTablesAsStringArray();
     }
 
+    @Override
+    public long getSecondsBehindSource() {
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReaderMetrics.java]: @gunnarmorling I think they are. MySQL provides only seconds precision and from the experience I've have from community the delay on order of seconds is extepected and accepted.

**Comment by jpechane** [comment]
@gunnarmorling Updated and force-pushed

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReaderMetrics.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReaderMetrics.java]: Changed to millis as discussed in the chat.

File: debezium-core/src/main/java/io/debezium/pipeline/source/spi/EventFromatter.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+public class EventFromatter {
+    private final String INDENT = "";
+
+    private Map<String, String> sourcePosition;
+    private Object key;
+    private Struct value;
+    private final StringBuilder string = new StringBuilder();
+
+    private void printStruct(Struct value, String indent) {
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/pipeline/source/spi/EventFromatter.java]: Thanks for the tip!

**Comment by jpechane** [comment]
@gunnarmorling Event formatter is polished

**Comment by gunnarmorling** [comment]
Applied. Great addition, thanks a lot, @jpechane!

**Pull request merged**